main/pong.go

```diff
@@ -0,0 +1,587 @@
+			ti = val
+		} else {
+			// New
+			ti.Last_seen = stratuxClock.Time
+		}
+		if mavLink.heading != 0 {
+			ti.Track = float32(mavLink.heading / 100)
+		} else {
+			signalLevelSimulated -= 5
+		}
+		ti.ReceivedMsgs += 1
+		ti.Icao_addr = mavLink.ICAO_address
+		ti.OnGround = false
+		ti.Addr_type = 0
+		if mavLink.squawk != 0 {
+			ti.Squawk = int(mavLink.squawk)
+		}
+		ti.TargetType = TARGET_TYPE_ADSB
+		if mavLink.emitter_type != 0 {
+			ti.Emitter_category = uint8(mavLink.emitter_type)
+		}
+		if mavLink.lat != 0 && mavLink.lon != 0 {
+			lat := float32(mavLink.lat) / 10000000.0
+			lng := float32(mavLink.lon) / 10000000.0
+			// Low signal may involve into a freeze location, update only if it really changes
+			if lng != ti.Lng && lat != ti.Lat {
+				ti.Lat = lat
+				ti.Lng = lng
+				if isGPSValid() {
+					lat := float64(mySituation.GPSLatitude)
+					lng := float64(mySituation.GPSLongitude)
+					ti.Distance, ti.Bearing = common.Distance(float64(lat), float64(lng), float64(ti.Lat), float64(ti.Lng))
+					ti.BearingDist_valid = true
+				} else {
+					ti.BearingDist_valid = false
+				}
+				ti.Position_valid = true
+				ti.ExtrapolatedPosition = false
+				ti.Last_seen = stratuxClock.Time
+				ti.Timestamp = time.Now().UTC()
+			}
+		} else {
+			ti.Position_valid = false
+			signalLevelSimulated -= 5
+		}
+		altitudeFoot := int32(float32(mavLink.altitude) / 304.8)
+		if altitudeFoot > 0 && altitudeFoot < 70000 {
+			// Low signal may involve into a freeze location, update only if it really changes
+			if ti.Alt != altitudeFoot {
+				ti.Alt = altitudeFoot
+				ti.Last_alt = stratuxClock.Time
+				ti.Last_seen = stratuxClock.Time
+				ti.Timestamp = time.Now().UTC()
+			}
+		} else {
+			signalLevelSimulated -= 5
+		}
+		if mavLink.altitude_type > 0 {
+			ti.AltIsGNSS = true
+		}
+		if mavLink.hor_velocity > 0 {
+			ti.Speed = uint16(float32(mavLink.hor_velocity) * 36 / 1852)
+			ti.Speed_valid = true
+			ti.Last_speed = stratuxClock.Time
+		} else {
+			ti.Speed_valid = false
+			signalLevelSimulated -= 5
+		}
+		ti.Vvel = int16(float32(mavLink.ver_velocity) * 60.0 / 10.0 / 3.048)
+		var callsignLen = 0
+		for a, b := range mavLink.callsign {
+			if b == ' ' || b == '\u0000' {
+				break
+			}
+			callsignLen = a + 1
+		}
+		thisReg, validReg := icao2reg(ti.Icao_addr)
+		if validReg {
+			ti.Reg = thisReg
+		}
+		if callsignLen > 0 {
+			ti.Tail = string(mavLink.callsign[:callsignLen])
+		} else {
+			signalLevelSimulated -= 5
+			if validReg {
+				ti.Tail = thisReg
+			}
+		}
+		// Timestamp and Last_seen are updated only if Location or Altitude changes
+		ti.NACp = 8
+		ti.NIC = 8
+		ti.Last_source = TRAFFIC_SOURCE_1090ES
+		ti.SignalLevel = float64(signalLevelSimulated)
+		postProcessTraffic(&ti)
+		traffic[ti.Icao_addr] = ti
+		registerTrafficUpdate(ti)
+		seenTraffic[ti.Icao_addr] = true
+		trafficMutex.Unlock()
+	}
+}
+
+func mavLinkParsePong(mavLinkFrame []byte) bool {
```

Comment:
Wait, pong can produce this natively too? The function never seems to be called though? Let's discuss that on discord.